bin/jenkins/common.sh

```diff
@@ -37,3 +37,18 @@ create_keypair(){
 
     rm -rf $tmpdir
 }
+
+safe_restart(){
+    local env=$1
+
+    . ./bin/secrets/common.sh
+
+    machine_name=$(vault_machine_name "$env")
+
+    setup_vault_addr "$machine_name"
+
+    token=$(vault read -field=value "secret/jenkins/config/admin_token")
+    master_ip=$(docker-machine ip "snappy-jenkins-${env}")
+
+    curl -u admin:"$token" -X POST "http://${master_ip}:8080/safeRestart" --data token="$token"
```

Comment:
What about printing a message here saying that jenkins will restart as soon as all the current jobs finish?